src/Service/Bulk/BulkCreateUsersService.php

```diff
@@ -22,297 +22,145 @@
-    private int $groupIndex = 0;
-
-    private string $generatedUsersFilePath;
-    private string $projectDir;
-
-    private const USER_CSV_HEAD = ['username', 'password', 'groupId'];
-    private const ASSIGNMENT_CSV_HEAD = ['username', 'lineItemSlug'];
-
-    private LineItemRepository $lineItemRepository;
     private AssignmentRepository $assignmentRepository;
-    private Filesystem $filesystem;
-    private GenerateGroupIdsService $generateGroupIdsService;
-    private CsvWriter $csvWriter;
-    private LtiInstanceRepository $ltiInstanceRepository;
-    private UserCreationResultMessage $userCreationMessage;
-    private LineItemCriteriaFactory $lineItemCriteriaFactory;
+    private AssigmentFactoryInterface $assigmentFactory;
+    private UserRepository $userRepository;
+    private StorageInterface $storage;
+    private UserPasswordEncoderInterface $hasher;
 
     private const DEFAULT_USERNAME_INCREMENT_VALUE = 0;
 
     public function __construct(
-        LineItemRepository $lineItemRepository,
         AssignmentRepository $assignmentRepository,
-        GenerateGroupIdsService $generateGroupIdsService,
-        CsvWriter $csvWriter,
-        LtiInstanceRepository $ltiInstanceRepository,
-        UserCreationResultMessage $userCreationMessage,
-        LineItemCriteriaFactory $lineItemCriteriaFactory,
-        Filesystem $filesystem,
-        string $generatedUsersFilePath,
-        string $projectDir
+        UserRepository $userRepository,
+        AssigmentFactoryInterface $assigmentFactory,
+        StorageInterface $storage,
+        UserPasswordEncoderInterface $hasher
     ) {
-        $this->lineItemRepository = $lineItemRepository;
         $this->assignmentRepository = $assignmentRepository;
-        $this->generateGroupIdsService = $generateGroupIdsService;
-        $this->userCreationMessage = $userCreationMessage;
-        $this->csvWriter = $csvWriter;
-        $this->generatedUsersFilePath = $generatedUsersFilePath;
-        $this->filesystem = $filesystem;
-        $this->ltiInstanceRepository = $ltiInstanceRepository;
-        $this->lineItemCriteriaFactory = $lineItemCriteriaFactory;
-        $this->projectDir = $projectDir;
+        $this->assigmentFactory = $assigmentFactory;
+        $this->storage = $storage;
+        $this->userRepository = $userRepository;
+        $this->hasher = $hasher;
     }
 
     /**
-     * @throws LineItemNotFoundException
+     * @param LineItem[] $lineItems
+     * @param string[] $userPrefixes
+     * @param GroupResolverInterface|null $groupResolver
      */
-    public function createUsers(
-        array $lineItemIds,
-        array $lineItemSlugs,
+    public function generate(
+        array $lineItems,
         array $userPrefixes,
         int $batchSize,
-        ?string $groupPrefix
-    ): UserCreationResult {
-        $notExistLineItemsArray = $userGroupIds = [];
+        ?GroupResolverInterface $groupResolver = null
+    ): void {
+        $data = date('Y-m-d');
```

Comment:
The variable name is confusing. It's not data but date.

Reply:
fixed